config/vite.json

```diff
@@ -8,7 +8,8 @@
     "autoBuild": true,
     "publicOutputDir": "vite-dev",
     "port": 3036,
-    "host": "127.0.0.1"
+    "host": "127.0.0.1",
+    "devServerConnectTimeout": 1.0
```

Comment:
A few things - the `500` you see here is a side effect of the [custom error handling](https://github.com/broadinstitute/single_cell_portal_core/pull/1339) we added a while back so we could have better logging/reporting and well-formed error responses in the API.  Really, these are `404`s - the client is making a `GET` for a module that `vite` is not rendering - but then our handling of this emits a secondary error.  This only happens in `development`, and is due to the fact that `ActionController::RoutingError` exceptions happen at the middleware level, not in our code, and Rails has very different handling of errors in `development` than in production-like environments.  But that's not the real problem - `vite` not rendering the module is the proximate cause.

But to your question re: timeout and Docker - yes, the two are independent.  I originally was just attempting to solve the above issue, but then had the thought that perhaps Docker could sidestep it altogether (Ubuntu's file handler limits are much higher by default than MacOS - `1048576` instead of `256`), and then pivoted to this avenue of work.  However, while it was _exceedingly_ rare, I did encounter the issue once while doing this work, so I left time timeout in as it does seem to address that particular problem.  It also doesn't really seem to affect performance in any meaningful way, at least that I can detect.

Reply:
Super insightful, thank you Jon!